### PR TITLE
Add libflatbuffers to rosdep/base.yaml

### PR DESCRIPTION
**Pull request by frankkusters**
Please add the following dependency to the rosdep database.

## Package name:

libflatbuffers

## Package Upstream Source:

https://github.com/google/flatbuffers

## Purpose of using this:

Flatbuffers is a serialization protocol by Google, similar to Protobuf.

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/search?suite=all&section=all&arch=any&searchon=names&keywords=libflatbuffers-dev
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/search?keywords=libflatbuffers-dev&searchon=names&exact=1&suite=all&section=all
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/flatbuffers/flatbuffers/
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/x86_64/flatbuffers/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-libs/flatbuffers
- macOS: https://formulae.brew.sh/
  - https://formulae.brew.sh/formula/flatbuffers
- Alpine: https://pkgs.alpinelinux.org/packages
  - https://pkgs.alpinelinux.org/packages?name=flatbuffers&branch=edge&repo=&arch=&maintainer=
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=24.05&show=flatbuffers&from=0&size=1&sort=relevance&type=packages&query=flatbuffers
- openSUSE: https://software.opensuse.org/package/
  - https://software.opensuse.org/package/flatbuffers
- rhel: https://rhel.pkgs.org/
  - https://pkgs.org/download/flatbuffers
- freebsd:
  - https://cgit.freebsd.org/ports/tree/devel/flatbuffers

### CODE REVIEW

**Review by wjwwood — APPROVED** [review]

File: rosdep/base.yaml [hunk 1]
```diff
@@ -3456,6 +3456,27 @@ libflann-dev:
   openembedded: [libflann@meta-ros-common]
   rhel: [flann-devel]
   ubuntu: [libflann-dev]
+libflatbuffers:
```
Comment (clalancette) [inline rosdep/base.yaml:3459]: To match the Ubuntu name, I think this should be:

```suggestion
libflatbuffers-dev:
```

File: rosdep/base.yaml [hunk 2]
```diff
@@ -3456,6 +3456,27 @@ libflann-dev:
...
+    '*': [flatbuffers]
+    '15.2': null
+    '15.3': null
+  osx:
+    homebrew:
+      packages: [flatbuffers]
+    macports:
+      packages: [flatbuffers]
+  rhel:
+    '*': [flatbuffers]
```
Comment (clalancette) [inline rosdep/base.yaml:3477]:
```suggestion
    '*': [flatbuffers-devel]
```

File: rosdep/base.yaml [hunk 3]
```diff
@@ -3456,6 +3456,27 @@ libflann-dev:
   openembedded: [libflann@meta-ros-common]
   rhel: [flann-devel]
   ubuntu: [libflann-dev]
+libflatbuffers:
+  alpine: [flatbuffers]
+  arch: [flatbuffers]
+  debian: [libflatbuffers-dev]
+  fedora: [flatbuffers]
```
Comment (clalancette) [inline rosdep/base.yaml:3463]:
```suggestion
  fedora: [flatbuffers-devel]
```

File: rosdep/base.yaml [hunk 4]
```diff
@@ -3456,6 +3456,27 @@ libflann-dev:
   openembedded: [libflann@meta-ros-common]
   rhel: [flann-devel]
   ubuntu: [libflann-dev]
+libflatbuffers:
+  alpine: [flatbuffers]
```
Comment (clalancette) [inline rosdep/base.yaml:3460]:
```suggestion
  alpine: [flatbuffers-dev]
```

File: rosdep/base.yaml [hunk 5]
```diff
@@ -3456,6 +3456,27 @@ libflann-dev:
...
+  arch: [flatbuffers]
+  debian: [libflatbuffers-dev]
+  fedora: [flatbuffers]
+  freebsd: [flatbuffers]
+  gentoo: [dev-libs/flatbuffers]
+  nixos: [flatbuffers]
+  opensuse:
+    '*': [flatbuffers]
```
Comment (clalancette) [inline rosdep/base.yaml:3468]:
```suggestion
    '*': [flatbuffers-devel]
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by frankkusters** [comment]
@clalancette I tested your proposed changes, and they look fine. My changes were based on the `libgrpc` key added in #37138. That key has the same issues. I opened an issue for it (#41602).

**Review by clalancette — APPROVED** [review]

**Pull request merged**